Oracle Tuxedo Dockerfiles for new version 22.1.0.0.0 (#2478)
oracle/docker-images: OracleTuxedo/core/README.md

# Tuxedo using Container Image

> Sample Container Image configurations to facilitate installation, configuration, and environment setup for DevOps users. This project includes Dockerfiles for Tuxedo 22.1.0.0.0 based on Oracle Linux and Oracle JDK 8 (Server).

Reviewer: ```suggestion
This sample container image is provided to facilitate installation, configuration and environment setup of Tuxedo 22.1.0.0.0. The image is based on Oracle Linux 7 and Oracle JDK 8 (ServerJRE).
```

The certification of Tuxedo on Container Image does not require the use of any file presented in this repository. Customers and users are welcome to use them as starters, and customize/tweak, or create from scratch new scripts and Dockerfiles.

> For pre-built images containing Oracle software, please check the [Oracle Container Registry](https://container-registry.oracle.com/).

Reviewer: ```suggestion
```

## Contents

> This folder contains the information and examples of how to use [Tuxedo](http://oracle.com/tuxedo) with Container Image

Reviewer: ```suggestion
This folder contains the information and examples of how to deploy [Tuxedo](http://oracle.com/tuxedo) in a container.
```

## How to build and run

> This project offers Dockerfiles for Tuxedo 22.1.0.0.0. To assist in building the images, you can use the buildContainerImage.sh script. See below for instructions and usage.
> 
> The buildContainerImage.sh script is just a utility shell script that performs MD5 checks and is an easy way for beginners to get started. Expert users are welcome to directly call container image build with their prefered set of parameters.

Reviewer: ```suggestion
To assist in building the images, the `buildContainerImage.sh` simplifies the process of ensuring the correct binary archive is available before starting the build process. More experienced users are welcome to build the image using their preferred container engine or CI/CD tool.
```

## Building Oracle JDK (Server JRE) base image

You must first download the Oracle Server JRE binary and drop in folder ../../OracleJava/java-8 and build that image. For more information, visit the [OracleJava](https://github.com/oracle/docker-images/blob/master/OracleJava) folder's [README](https://github.com/oracle/docker-images/blob/master/OracleJava/README.md) file.

    cd ../../OracleJava/java-8

>     sh build.sh

Reviewer: ```suggestion
The Tuxedo image uses the Oracle JDK 8 (Server JRE) container image as its base. Please follow the [Oracle Java image](https://github.com/oracle/docker-images/blob/master/OracleJava) documentation to build that image before continuing.

> ## Building Tuxedo Container Image Install Images

Reviewer: ```suggestion
## Building Tuxedo Install Images
```

1. Into an empty directory:

>    * Download the Tuxedo 22.1.0.0.0 Linux 64bit installer from [OTN](http://www.oracle.com/technetwork/middleware/tuxedo/downloads/index.html)

Reviewer: ```suggestion
   * Download the [Tuxedo 22.1.0.0.0 Linux 64-bit installer](http://www.oracle.com/technetwork/middleware/tuxedo/downloads/index.html) from the Oracle Technology Network.
```

>    * Download all the files from this GitHub repository

Reviewer: ```suggestion
   * Clone or download this repository
```

>    * Drop the downloaded Tuxedo installer to the corresponding version directory

Reviewer: ```suggestion
   * Move the downloaded Tuxedo installer to the corresponding version directory
```

   * Optionally download the latest Tuxedo rolling patch from My Oracle Support

> 2. cd dockerfiles

Reviewer: ```suggestion
2. Change to the `OracleTuxedo/core/dockerfiles` directory in your local copy of the repository
```

> 3. Execute './buildContainerImage.sh' to show the usage of the command. Follow [the guide](./dockerfiles/README.md) to create a Container image.

Reviewer: ```suggestion
3. Run './buildContainerImage.sh' to see the available parameters for this script then follow [the documentation](./dockerfiles/README.md) to build the image.
```

> You should end up with a Container image tagged oracle/tuxedo:version, for instance, oracle/tuxedo:22.1.0.0.0 and oracle/tuxedo:latest

Reviewer: ```suggestion
This process builds a container image image named `oracle/tuxedo` tagged by version. For example, `oracle/tuxedo:22.1.0.0.0` and `oracle/tuxedo:latest`.
```

> You can then start the image in a new container with: ``docker run -d -v \${LOCAL_DIR}:/u01/oracle/user_projects oracle/tuxedo:<VERSION>``.

Reviewer: ````suggestion
You can then start a container based on this image using the following command:

```shell
podman run -d -v "${LOCAL_DIR}:/u01/oracle/user_projects" oracle/tuxedo:22.1.0.0.0
```
````

> Note: \${LOCAL_DIR} is a local dir which used in Container image as external storage, it can be any dir.

Reviewer: ```suggestion
Note: `${LOCAL_DIR}` must resolve to a local directory in which the container can store data.
```

## Tuxedo Distribution and Documentation

> * For more information on the Tuxedo 22.1.0.0.0 Distribution, visit [Tuxedo Installer](http://www.oracle.com/technetwork/middleware/tuxedo/downloads/index.html).

Reviewer: ```suggestion
* For more information on Tuxedo 22.1.0.0.0, visit [Tuxedo Installer](http://www.oracle.com/technetwork/middleware/tuxedo/downloads/index.html).
```

## License

To download and run Tuxedo 22.1.0.0.0 regardless of inside or outside a Container container, you must download the binaries from Oracle website and accept the license indicated at that page.

> All scripts and files hosted in this project and GitHub [docker/OracleTuxedo](./) repository required to build the Container images are, unless otherwise noted, released under [UPL 1.0](https://oss.oracle.com/licenses/upl/) license.

Reviewer: ```suggestion
All scripts and files hosted in this repository required to build the container images are, unless otherwise noted, released under [UPL 1.0](https://oss.oracle.com/licenses/upl/) license.
```
